### PR TITLE
test(e2e benchmark): refactors E2EThroughput

### LINKED ISSUE #3363: Consolidate test settings for enhanced e2e benchmarking test scenario management
In the e2e benchmarking tests,  the test scenario is defined by a core set of parameters that influence the performance evaluation of the experiments. These parameters vary in nature; some are configuration settings, some are consensus/p2p parameters, and others define the network topology, etc. Consolidating these parameters in a single struct and ensuring that each testnet is associated with one and use that  when setting up nodes, configurations, and parameters will simplify the process for test developers. This also facilitates the comparison of different test scenarios. 

This issue aims to perform the necessary refactoring to achieve this.

### PR DESCRIPTION
**Pull request by staheri14**
A follow-up PR for #3391 that addresses and closes #3363.

Summary of changes:
The `BenchmarkTest` type has been introduced, inheriting all the behaviors of a `Testnet`. Additionally, it encapsulates a `Manifest` that defines the test network topology, parameters, and configurations, etc., and directs the network formation.

### CODE REVIEW

File: test/e2e/benchmark/benchmark.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	// create tx clients and point them to the validators
+	log.Println("Creating tx clients")
+
+	err = b.CreateTxClients(b.manifest.TxClientVersion,
+		b.manifest.BlobSequences,
+		b.manifest.BlobSizes,
+		b.manifest.TxClientsResource, gRPCEndpoints)
+	testnet.NoError("failed to create tx clients", err)
+
+	// start the testnet
```
Comment (rootulp) [inline test/e2e/benchmark/benchmark.go:52]: Do the following lines just setup the testnet or actually start it? I think just setup so
```suggestion
	// Set up the testnet
```

which IMO the comment isn't necessary given there is a print statement that describes it below

File: test/e2e/benchmark/throughput.go [hunk 2]
```diff
@@ -50,56 +50,23 @@ func E2EThroughput() error {
...
-	log.Println("Starting testnet")
-	testnet.NoError("failed to start testnet", testNet.Start())
-
-	// once the testnet is up, start the tx clients
-	log.Println("Starting tx clients")
-	testnet.NoError("failed to start tx clients", testNet.StartTxClients())
-
-	// wait some time for the tx clients to submit transactions
-	time.Sleep(manifest.TestDuration)
+	testnet.NoError("failed to run the bench test", benchTest.Run())
```
Comment (rootulp) [inline test/e2e/benchmark/throughput.go:64]: [nit] bench => benchmark in all of these lines

**Review by rootulp — DISMISSED** [review]

File: test/e2e/benchmark/throughput.go (hunk 2 above)
Comment (staheri14) [reply test/e2e/benchmark/throughput.go]: Sure, done https://github.com/celestiaorg/celestia-app/pull/3395/commits/62a8781f9c4db99c45247724d076e6a5d4a4485e

File: test/e2e/benchmark/benchmark.go (hunk 1 above)
Comment (staheri14) [reply test/e2e/benchmark/benchmark.go]: Right, it sets up the testnet, fixed it in https://github.com/celestiaorg/celestia-app/pull/3395/commits/62a8781f9c4db99c45247724d076e6a5d4a4485e

**Review by rootulp — APPROVED** [review]

**Review by rach-id — APPROVED** [review]
utAck

**Pull request merged**